packages/atlas/src/providers/reactionsTransactions/provider.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useMemo, useState } from 'react'
+
+import { CommentFieldsFragment } from '@/api/queries'
+
+type ContextValue = {
+  commentInputIsProcessingCollection: Set<string>
+  setCommentInputIsProcessing: ({ commentInputId, value }: { commentInputId: string; value: boolean }) => void
+  videoReactionProcessing: boolean
+  setVideoReactionProcessing: React.Dispatch<React.SetStateAction<boolean>>
+  processingCommentReactionId: string | null
+  setProcessingCommentReactionId: React.Dispatch<React.SetStateAction<string | null>>
+  originalComment: CommentFieldsFragment | null
+  setOriginalComment: React.Dispatch<React.SetStateAction<CommentFieldsFragment | null>>
+  highlightedComment: string | null
+  setHighlightedComment: React.Dispatch<React.SetStateAction<string | null>>
+  showEditHistory: boolean
+  setShowEditHistory: React.Dispatch<React.SetStateAction<boolean>>
+  isEditingCommentCollection: Set<string>
+  setIsEditingComment: ({ commentId, value }: { commentId: string; value: boolean }) => void
+}
+
+export const ReactionsTransactionsContext = React.createContext<ContextValue | undefined>(undefined)
+ReactionsTransactionsContext.displayName = 'ReactionsTransactionsContext'
+
+export const ReactionsTransactionsProvider: React.FC = ({ children }) => {
+  const [videoReactionProcessing, setVideoReactionProcessing] = useState(false)
+  const [processingCommentReactionId, setProcessingCommentReactionId] = useState<string | null>(null)
+  const [originalComment, setOriginalComment] = useState<CommentFieldsFragment | null>(null)
+  const [highlightedComment, setHighlightedComment] = useState<string | null>(null)
```

Comment:
I think this should still be kept at `CommentsSection` level, it doesn't make much sense for this to be global. Every comment/input can have some event that is emitted when highlight should be changed. Example how `CommentsSection` could look like:
```tsx
const [highlightedComment, setHighlightedComment] = useState<string | null>(null)

// timeout for highlight
useEffect(/* ... */)

return (
  <div>
    <CommentInput onCreated={comment => setHighlightedComment(comment.id)} />
    {comment.map(comment => <Comment {...comment} onEdited={() => setHighlightedComment(comment.id)} />)}
  </div>
)
```